apps/laboratory/src/components/RandomLink.tsx

```diff
@@ -7,7 +7,7 @@ export function RandomLink({ hrefs, children }: { hrefs: string[]; children: Rea
   const [href, setHref] = useState<string>()
   useEffect(() => {
     const newHref = hrefs[Math.floor(Math.random() * hrefs.length)]
-    if (!href) {
+    if (!newHref) {
```

Comment:
can we do this as its an array `hrefs.length === 0`

Reply:
do we wanted to check the hrefs passed to the component? right

Reply:
This wouldn't assert that `newHref` is not-`undefined` because array indexes return `| undefined`. Removed the check entirely since `href` can be `undefined`

Reply:
>do we wanted to check the hrefs passed to the component? right
The original goal was to avoid the type error assigning `string | undefined` to `Link[href]` which is `string`

Reply:
I think this will fix it
```
export function RandomLink({ hrefs, children }: { hrefs: string[]; children: React.ReactNode }) {
  const [href, setHref] = useState<string>()
  useEffect(() => {
    if (hrefs.length > 0) {
      setHref(hrefs[Math.floor(Math.random() * hrefs.length)])
    }
  }, [hrefs])

  return href ? <Link href={href}>{children}</Link> : <></>
}

```
we need to check `hrefs.length > 0` before setting  ```setHref(hrefs[Math.floor(Math.random() * hrefs.length)])```